scripts/e2eTest.js

```diff
@@ -0,0 +1,170 @@
+    process.exit(1);
+});
+process.on('uncaughtException', (error) => {
+    console.error('[ERROR] Uncaught Exception:', error);
+    process.exit(1);
+});
+
+/**
+ * Main orchestration function.
+ */
+async function main() {
+    ensureRequiredEnvVars();
+
+    console.log('[INFO] Starting E2E testing sequence...');
+    console.log('[INFO] Deploying code version...');
+    await deployCode();
+
+    // We test two different record models:
+    const recordModels = ['master-level', 'variation-level'];
+
+    const indexPrefixes = {
+        'master-level': 'basex',
+        'variation-level': 'varx'
+    };
+
+    for (const model of recordModels) {
+        console.log(`\n[INFO] === Running E2E for: ${model} ===`);
+        process.env.RECORD_MODEL = model;
+        process.env.INDEX_PREFIX = indexPrefixes[model];
+
+        console.log('[INFO] Importing site preferences...');
+        await importPreferences();
+
+        console.log('[INFO] Running SFCC job...');
+        await runSFCCJob();
+
+        console.log('[INFO] Running Variation Index Tests (Jest)...');
+        await runVariationIndexTests();
+
+        console.log('[INFO] Running Cypress Frontend Tests...');
+        await runCypressTests();
+    }
+
+    console.log('\n[INFO] All E2E tests completed successfully.');
+    process.exit(0);
+}
+
+/**
+ * Ensures that essential environment variables are set.
+ */
+function ensureRequiredEnvVars() {
+    const requiredVars = [
+        'TEST_PRODUCT_ID',
+        'TEST_MASTER_PRODUCT_ID',
+        'CODE_VERSION',
+        'SANDBOX_HOST',
+        'SFCC_OAUTH_CLIENT_ID',
+        'SFCC_OAUTH_CLIENT_SECRET',
+        'ALGOLIA_APP_ID',
+        'ALGOLIA_API_KEY'
+    ];
+
+    requiredVars.forEach((envVar) => {
+        if (!process.env[envVar]) {
+            throw new Error(`Missing required environment variable: ${envVar}`);
+        }
+    });
+}
+
+/**
+ * Runs the Variation Index Tests via Jest's programmatic API.
+ */
+async function runVariationIndexTests() {
+    // We can specify more advanced config if needed:
+    const config = {
+        runInBand: true,        // run tests sequentially
+        testPathPattern: ['test/e2e/algolia_index.test.js']
+    };
+    const { results } = await runCLI(config, [process.cwd()]);
+    if (results.numFailedTests || results.numFailedTestSuites) {
+        throw new Error('Variation Index Tests failed. Please check the Jest results above.');
+    }
+}
+
+/**
+ * Runs the Cypress End-to-End Tests via the Cypress module API.
+ */
+async function runCypressTests() {
+    const cypressEnv = {
+        RECORD_MODEL: process.env.RECORD_MODEL,
+    };
+
+    const result = await cypress.run({
+        spec: 'cypress/e2e/frontend.cy.js',
+        env: cypressEnv,
+        browser: 'chrome',
+        headless: true
+    });
+
+    if (result.totalFailed > 0) {
+        throw new Error('Cypress tests failed. Please check logs above.');
+    }
```

Comment:
I found a problem with that check.
I don't have Chrome on my machine, so the tests weren't run but the whole E2E was reported as success. This is because `results` contains the following:
```
{
  status: 'failed',
  failures: 1,
  message: 'Could not find Cypress test run results'
}
```
Cypress' documentation explains in details how to properly handle errors: https://docs.cypress.io/app/references/module-api#Handling-errors


Note: it was not easy to spot the problem, because logs are flooded by the "Job status result" logs that are too verbose. Can you improve it to display only the relevant information?

Reply:
I updated here like that:

```
    console.log(`[INFO] Cypress run result: status=${result.status}, totalFailed=${result.totalFailed || 0}, failures=${result.failures || 0}`);

    if (result.status === 'failed' || result.failures > 0 || (result.totalFailed && result.totalFailed > 0)) {
        throw new Error(`Cypress tests failed with status: ${result.status}, failures: ${result.failures}, message: ${result.message}`);
    }
```

Reply:
Hmm ok. Note that when you have only a few tests that fail, all those are undefined so you error message will look like this:
> Cypress tests failed with status: undefined, failures: undefined, message: undefined

But ok at least there is no false positive anymore.
You've also missed my last sentence:
> Can you improve the verbosity of the "Job status result" logs to display only the relevant information?

Since this PR is only meant to improve the tests, it's the good occasion to do it.